### PR TITLE
Add StackEnvironment

### PR DESCRIPTION
**Pull request by ChristianWieck**
This PR:
- Add a new `StackEnvironment` to combine two (or more) `OfflineEnviroments` vertically
- Add helper methods and syntactic sugar to the `OfflineEnvironment`
- Add tests for the new environment

### CODE REVIEW

File: src/flowcean/core/environment/stack.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        """Get data from the environment.
+
+        Returns:
+            The loaded dataset.
+        """
+        dfs = [env.get_data() for env in self.environments]
+        return reduce(
+            lambda x, y: x.vstack(y),
+            dfs,
+        )
```
Comment (schmidma) [inline src/flowcean/core/environment/stack.py:36]: There is [`pl.concat`](https://docs.pola.rs/api/python/stable/reference/api/polars.concat.html)

File: src/flowcean/core/environment/offline.py [hunk 2]
```diff
@@ -35,3 +35,28 @@ def as_stream(self, batch_size: int = 1) -> StreamingOfflineData:
...
+    def __and__(self, other: OfflineEnvironment) -> OfflineEnvironment:
+        """Combine this environment with another one vertically.
+
+        Args:
+            other: The environment to append vertically.
+
+        Returns:
+            The combined environment.
+        """
+        return self.stack(other)
```
Comment (schmidma) [inline src/flowcean/core/environment/offline.py:62]: I'm not sure about that. This is something like `a_env + b_env`, correct? This syntax is not showing whether it concatenates them vertically or horizontally. This syntax could also combine features.

File: src/flowcean/core/environment/stack.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+from .offline import OfflineEnvironment
+
+
+class StackEnvironment(OfflineEnvironment):
+    """Combines multiple OfflineEnvironments into a single one.
+
+    This environment combines multiple OfflineEnvironments into a single one by
+    stacking them vertically. All environments need to share the identical
+    features.
```
Comment (schmidma) [inline src/flowcean/core/environment/stack.py:14]: The fact that they must share identical feature names is only checked in the end, when the data is actually requested. Do we want to do that earlier? Or is this okay for now?

**Review by schmidma — CHANGES_REQUESTED** [review]
minor questions

File: src/flowcean/core/environment/stack.py (hunk 1 above)
Comment (ChristianWieck) [reply src/flowcean/core/environment/stack.py:36]: Forgot about that, thanks

File: src/flowcean/core/environment/offline.py (hunk 2 above)
Comment (ChristianWieck) [reply src/flowcean/core/environment/offline.py:62]: The syntax is `env_a & env_b`. The general idea was to use `|` for horizontal concatenation of two frames. However, I understand that this is not clear. I would suggest to remove the feature for now and have a discussion about it in one of the next meetings?

File: src/flowcean/core/environment/stack.py (hunk 3 above)
Comment (ChristianWieck) [reply src/flowcean/core/environment/stack.py:14]: This boils down to a general question of error handling: Should we wait and "let the error happen"/polars tell us we did something wrong, or should we check when an environment/transformation is created and throw our own errors? Currently we mostly use the first approach. I'm open to both approaches as long as we keep one through the framework.

File: src/flowcean/core/environment/offline.py (hunk 2 above)
Comment (schmidma) [reply src/flowcean/core/environment/offline.py:62]: Agreed, sounds good

File: src/flowcean/core/environment/stack.py (hunk 3 above)
Comment (schmidma) [reply src/flowcean/core/environment/stack.py:13]: Let's keep it like this for now...

File: src/flowcean/core/environment/offline.py (hunk 2 above)
Comment (ChristianWieck) [reply src/flowcean/core/environment/offline.py]: I created the issue #89 to keep track of further discussions.

**Review by schmidma — APPROVED** [review]

**Pull request merged**